Green-Kittens/bathroom-finder-frontend
# Create a review on Review Form

# Checklist

- [ ] Have I clearly explained the issue or feature request that this PR addresses?
- [ ] Have I provided where new implementations for the PR have been made?
- [ ] Did I highlight any areas that may need special attention during the review?
- [ ] If applicable, did I specify any setup requirements or dependencies for testing?
- [ ] Have I disclosed any known issues or limitations related to this PR? Did I explain why these limitations exist and how they might be addressed in the future?
- [ ] Have I provided enough context for someone unfamiliar with the issue or the codebase to understand the changes?
- [ ] Have I assigned appropriate reviewers who are knowledgeable about the changes? Did I consider including team members who might be affected or interested in the changes?

# Description

## Summary
Issue #146:
- Original implementation did not pass review form inputs to backend to create a review for a facility. 
- New implementation now passes review form inputs to backend to create a review for a facility. 

## Expectations
- Steps for Testing: 
   1. Press any facility marker on the main map 
   2. Press the option to create a review for the facility you are currently on
   3. Fill out the review form
   4. If the star rating field is filled out (greater than zero), upon pressing the submit button, expect to navigate back to the main screen
   5. Click back into the facility marker you just created a review for, check the console log for two results:
       - a list of reviews fetched from the getAllReviews controller function 
       - the FacilityID of the current facility 
   6. Check to see if your review has been added to the list of reviews fetched from getAllReviews. The FacilityID of the review should match the id of the facility profile you are on

## Additional Comments/Notes
- Once Kayla's PR is merged, your review should render, but for now check the console to make sure that your review is being added to the db
- SIDE NOTE: getAllReviews is suppose to get all reviews for a specific FacilityID but returns all existing reviews in the db. The result of getAllReviews contains reviews with different facilityIDs, shouldn't it only return the reviews with facilityIDs that match the one being passed into the function?
    - this will affect Kayla's issue for rendering the reviews on the reviews page as getAllReviews doesn't seem to be returning the correct list of reviews
